runtime/near-vm/engine/src/universal/code_memory.rs

```diff
@@ -260,7 +255,11 @@ impl LimitedMemoryPool {
     /// Get a memory mapping, at least `size` bytes large.
     pub fn get(&self, size: usize) -> rustix::io::Result<CodeMemory> {
         let mut guard = self.pool.lock().expect("unreachable due to panic=abort");
-        let mut memory = guard.pop().ok_or(rustix::io::Errno::NOMEM)?;
+        let mut memory = match guard.pop() {
+            Some(m) => m,
+            // This memory will later return to this pool via the drop of `CodeMemory`.
```

Comment:
If mostly `preallocate_count` is sufficient and additional memories are required only during spikes, would it be an option to unmap instead of returning to the pool?

Reply:
There is no such thing as a spike, except maybe in RPC nodes that execute view calls. We just aren't able to access configuration variables that would let us determine the exact number of mappings anywhere near where this pool is getting constructed.